contracts/interfaces/IInterchainMultisig.sol

```diff
@@ -0,0 +1,49 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.0;
+
+import { IBaseWeightedMultisig } from './IBaseWeightedMultisig.sol';
+import { ICaller } from './ICaller.sol';
+
+/**
+ * @title IMultisig Interface
+ * @notice This interface extends IMultisigBase by adding an execute function for multisignature transactions.
+ */
+interface IInterchainMultisig is ICaller, IBaseWeightedMultisig {
+    error NotSelf();
+    error InvalidProof();
+    error AlreadyExecuted();
+    error InvalidPayloadType();
+    error InvalidChainNameHash();
+    error InvalidTarget();
+    error InvalidRecipient();
+
+    struct InterCall {
+        bytes32 chainNameHash;
```

Comment:
string chain would be nicer for debugging. gas optimization is less important for this contract